framework/src/modules/chain/bft/headers_list.js

```diff
@@ -112,6 +112,10 @@ class HeadersList {
 	top(size) {
 		assert(size, 'Please provide the size');
 
+		if (this.length <= size) {
+			return [...this._items];
+		}
+
```

Comment:
It was a very minor logical error, should have been catched in development/review cycle.  🤦‍♂